### PR TITLE
Fix NullPointerException

### LINKED ISSUE #17: NullPointerException with null data in adapter
My data in my adapter is empty when the view is loaded and I get the following exception.

 java.lang.NullPointerException: Attempt to invoke virtual method 'android.view.ViewParent android.view.View.getParent()' on a null object reference
            at android.support.v7.widget.RecyclerView.getChildViewHolder(RecyclerView.java:2567)
            at xyz.danoz.recyclerviewfastscroller.calculation.progress.VerticalLinearLayoutManagerScrollProgressCalculator.calculateScrollProgress(VerticalLinearLayoutManagerScrollProgressCalculator.java:29)
            at xyz.danoz.recyclerviewfastscroller.AbsRecyclerViewFastScroller.onLayout(AbsRecyclerViewFastScroller.java:202)
            at android.view.View.layout(View.java:15596)
            at android.view.ViewGroup.layout(ViewGroup.java:4966)

The problem is in VerticalLinearLayoutManagerScrollProgressCalculator:

    @Override
    public float calculateScrollProgress(RecyclerView recyclerView) {
        LinearLayoutManager layoutManager = (LinearLayoutManager) recyclerView.getLayoutManager();
        int lastFullyVisiblePosition = layoutManager.findLastCompletelyVisibleItemPosition();

        View visibleChild = recyclerView.getChildAt(0);
        ViewHolder holder = recyclerView.getChildViewHolder(visibleChild);

There is no child at index 0.

This method should handle the case when the RecyclerView has no children, such as when there is not yet any data.

### PR DESCRIPTION
**Pull request by Maxr1998**
fix #17 by returning 0 as progress when adapter contains no items

### CODE REVIEW

File: recyclerviewfastscroller/src/main/java/xyz/danoz/recyclerviewfastscroller/calculation/progress/VerticalLinearLayoutManagerScrollProgressCalculator.java [hunk 1]
```diff
@@ -26,6 +26,8 @@ public float calculateScrollProgress(RecyclerView recyclerView) {
         int lastFullyVisiblePosition = layoutManager.findLastCompletelyVisibleItemPosition();
 
         View visibleChild = recyclerView.getChildAt(0);
+        if (visibleChild == null)
+            return 0;
```
Comment (danoz73) [inline recyclerviewfastscroller/src/main/java/xyz/danoz/recyclerviewfastscroller/calculation/progress/VerticalLinearLayoutManagerScrollProgressCalculator.java]:
```java
if (visibleChild == null) {
   return 0;
}
```
Thanks!

**Comment by danoz73** [comment]
Just a small nit. Lmk when you update!

File: recyclerviewfastscroller/src/main/java/xyz/danoz/recyclerviewfastscroller/calculation/progress/VerticalLinearLayoutManagerScrollProgressCalculator.java (hunk 1 above)
Comment (Maxr1998) [inline recyclerviewfastscroller/src/main/java/xyz/danoz/recyclerviewfastscroller/calculation/progress/VerticalLinearLayoutManagerScrollProgressCalculator.java]: Why would you add the extra brackets? It'd work without them, but you can still add them yourself. No need for extra PR imho..

Comment (danoz73) [inline recyclerviewfastscroller/src/main/java/xyz/danoz/recyclerviewfastscroller/calculation/progress/VerticalLinearLayoutManagerScrollProgressCalculator.java]: https://source.android.com/source/code-style.html#use-standard-brace-style

Comment (Maxr1998) [inline recyclerviewfastscroller/src/main/java/xyz/danoz/recyclerviewfastscroller/calculation/progress/VerticalLinearLayoutManagerScrollProgressCalculator.java]: Ahh, interesting. But it's only recommended/forced for android source. I will still change it, wait a moment..

**Comment by Maxr1998** [comment]
Done.

**Comment by IlyaEremin** [comment]
@danoz73 merge plz

**Pull request closed without merging**